contributors/design-proposals/local-storage-pv.md

```diff
@@ -0,0 +1,631 @@
+root directory.  The contents are deleted when a pod dies.  Limitations:
+
+  * Volume lifetime is bound to the pod lifetime.  Pod failure is more likely
+than node failure, so there can be increased network and storage activity to
+recover data via replication and data backups when a replacement pod is started.
+  * Multiple disks are not supported unless the administrator aggregates them
+into a spanned or RAID volume.  In this case, all the storage is shared, and
+IOPS guarantees cannot be provided.
+  * There is currently no method of distinguishing between HDDs and SDDs.  The
+“medium” field could be expanded, but it is not easily generalizable to
+arbitrary types of mediums.
+
+* HostPath: A direct mapping to a specified directory on the node.  The
+directory is not managed by the cluster.  Limitations:
+
+  * Admin needs to manually setup directory permissions for the volume’s users.
+  * Admin has to manage the volume lifecycle manually and do cleanup of the data and
+directories.
+  * All nodes have to have their local storage provisioned the same way in order to
+use the same pod template.
+  * There can be path collision issues if multiple pods get scheduled to the same
+node that want the same path
+  * If node affinity is specified, then the user has to do the pod scheduling
+manually.
+
+* Provider’s block storage (GCE PD, AWS EBS, etc): A remote disk that can be
+attached to a VM instance.  The disk’s lifetime is independent of the pod’s
+lifetime.  Limitations:
+
+  * Doesn’t meet performance requirements.
+[Performance benchmarks on GCE](https://cloud.google.com/compute/docs/disks/performance)
+show that local SSD can perform better than SSD persistent disks:
+
+    * 16x read IOPS
+    * 11x write IOPS
+    * 6.5x read throughput
+    * 4.5x write throughput
+
+* Networked filesystems (NFS, GlusterFS, etc): A filesystem reachable over the
+network that can provide shared access to data.  Limitations:
+
+  * Requires more configuration and setup, which adds operational burden and
+cost.
+  * Requires a high performance network to achieve equivalent performance as
+local disks, especially when compared to high-performance SSDs.
+
+Due to the current limitations in the existing volume types, a new method for
+providing persistent local storage should be considered.
+
+## Feature Plan
+
+A detailed implementation plan can be found in the
+[Storage SIG planning spreadhseet](https://docs.google.com/spreadsheets/d/1t4z5DYKjX2ZDlkTpCnp18icRAQqOE85C1T1r2gqJVck/view#gid=1566770776).
+The following is a high level summary of the goals in each phase.
+
+### Phase 1
+
+* Support Pod, Deployment, and StatefulSet requesting local storage
+* Support pre-configured, statically partitioned, filesystem-based local storage
+
+### Phase 2
+
+* Block devices and raw partitions
+* Smarter PV binding to consider local storage and pod scheduling constraints
+
+### Phase 3
+
+* Support common partitioning patterns
+* Volume forgiveness
```

Comment:
what is this? needs more details or link to another doc?

Reply:
This is the tolerations idea I mentioned above.  I'm leaving the design details to a future release when we actually have time to work on it.  There's too many features needed for local volumes to design it all in one go.

Reply:
thanks.